src/hotspot/share/gc/shared/genArguments.cpp

```diff
@@ -306,7 +308,8 @@ void GenArguments::initialize_size_info() {
       initial_old_size = MaxOldSize;
     }
 
-    MinOldSize = MIN2(initial_old_size, MinHeapSize - MinNewSize);
+    // Make sure MinOldSize <= OldSize
+    MinOldSize = MIN2(MinOldSize, initial_old_size);
```

Comment:
Do you have a test case that requires this? I'd expect `MIN3` above to be enough.

Reply:
Without this, gtest fails on MacOSX (https://github.com/zhengyu123/jdk/actions/runs/8743941793/job/23997262102)